modules/openapi-generator/src/main/resources/typescript/types/ObservableAPI.mustache

```diff
@@ -61,19 +62,52 @@ export class Observable{{classname}} {
      * @param {{#required}}{{paramName}}{{/required}}{{^required}}[{{paramName}}]{{/required}}{{#description}} {{description}}{{/description}}
      {{/allParams}}
      */
-    public {{nickname}}WithHttpInfo({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}_options?: Configuration): Observable<HttpInfo<{{{returnType}}}{{^returnType}}void{{/returnType}}>> {
-        const requestContextPromise = this.requestFactory.{{nickname}}({{#allParams}}{{paramName}}, {{/allParams}}_options);
+    public {{nickname}}WithHttpInfo({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}_options?: Configuration{{^useInversify}}Options{{/useInversify}}): Observable<HttpInfo<{{{returnType}}}{{^returnType}}void{{/returnType}}>> {
+	let _config = this.configuration
+	let allMiddleware: Middleware[] = []
+	{{#useInversify}}
+	if (_options){
+	    _config = _options
+	}
+	allMiddleware = _config?.middleware
+	{{/useInversify}}
+	{{^useInversify}}
+	if (_options && _options.middleware){
+	    // call-time middleware provided
+            let calltimeMiddleware: Middleware[] = _options.middleware
 
+	    switch(_options.middlewareMergeStrategy){
+	    	case 'append':
+                    allMiddleware = this.configuration.middleware.concat(calltimeMiddleware)
+		    break;
+	    	case 'prepend':
+		    allMiddleware = calltimeMiddleware.concat(this.configuration.middleware)
+		    break;
+		default: // replace
+		    allMiddleware = calltimeMiddleware
+	    }
+	}
+	if (_options){
+		_config = {
+		    baseServer: _options.baseServer || this.configuration.baseServer,
+		    httpApi: _options.httpApi || this.configuration.httpApi,
+		    authMethods: _options.authMethods || this.configuration.authMethods,
+		    middleware: allMiddleware || this.configuration.middleware
+		}
```

Comment:
I might be missing something or could this be:

```suggestion
                _config = {
                  ...this.configuration,
                  ..._options,
                  middleware: allMiddleware || this.configuration.middleware,
                };
```

when inversify is on, we have `_config = _options` as well, which suggests these two objects are interchangeable in their shape (without checking)

Reply:
i tried this out, but some `undefined` fields were still making it through instead of being preferred from the configuration